sm2/src/dsa.rs

```diff
@@ -197,3 +204,20 @@ impl TryFrom<&[u8]> for Signature {
         Signature::from_slice(bytes)
     }
 }
+
+#[cfg(all(feature = "alloc", feature = "pkcs8"))]
+impl SignatureBitStringEncoding for Signature {
+    fn to_bitstring(&self) -> der::Result<der::asn1::BitString> {
+        der::asn1::BitString::new(0, self.to_vec())
+    }
+}
+
+#[cfg(all(feature = "alloc", feature = "pkcs8"))]
+impl AssociatedAlgorithmIdentifier for Signature {
+    type Params = AnyRef<'static>;
+
+    const ALGORITHM_IDENTIFIER: AlgorithmIdentifierRef<'static> = AlgorithmIdentifierRef {
+        oid: <Sm2 as AssociatedOid>::OID,
```

Comment:
Looks like this needs to be a slightly different OID (ending in 501 instead of 301)